lib/bm-response.js

```diff
@@ -0,0 +1,58 @@
+/* @flow */
+'use strict'
+
+/* ::
+import type {BmRequest, Headers} from './wrapper.js'
+*/
+
+const privateVars = new WeakMap()
+
+class BmResponse {
+  constructor (
+    request /* : BmRequest */
+  ) {
+    privateVars.set(this, {
+      headers: {},
+      payload: undefined,
+      request,
+      statusCode: 200
+    })
+  }
+
+  get headers () /* : Headers */ {
+    return privateVars.get(this).headers
```

Comment:
If preventing direct access to private values is important, then you should not return the internal "headers" object directly. Rather, you should return a copy of it, so that consumers are unable to modify the internal / canonical copy. e.g. `return Object.assign({}, privateVars.get(this).headers)`